### DISCUSSION TITLE
Cannot connect to SQL Server database

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#5241] [Q&A]
Hi,

I am trying to connect Mirth 3.12 to a SQL Server database, to no avail so far. The Mirth Connect Service is running.
I am supposed to connect to the database using Windows authentication, and when I do so in SQL Server Management Studio, I am able to manipulate the database just fine. 
The key line in mirth.log is, I think:
```
java.lang.RuntimeException: Driver net.sourceforge.jtds.jdbc.Driver claims to not accept jdbcUrl, jdbc:sqlserver://localhost:1433;databaseName=MOSOS_Mirth
```
which suggests the url is wrongly constructed. So I have tried various options, inspired by both [https://docs.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver16](url) and [https://forums.mirthproject.io/forum/mirth-connect/support/11756-mirth-connect-sqlsvr2008r2-w-windows-authentication-only](url), but alas:
```
java.lang.RuntimeException: Driver net.sourceforge.jtds.jdbc.Driver claims to not accept jdbcUrl, jdbc:sqlserver://localhost:1433;databaseName=MOSOS_Mirth;integratedSecurity=true
java.lang.RuntimeException: Driver net.sourceforge.jtds.jdbc.Driver claims to not accept jdbcUrl, jdbc:jtds.sqlserver://hostname.domain.com:1433/MOSOS_Mirth;useNTLMv2=true
```
If more info is necessary - more logging perhaps - I am of course perfectly happy to provide it.
Anyone any suggestions? Thanks in advance!

**EDIT**
Can perhaps the cause be that both ports 8080 and 8443 - the default server ports that I indeed use - do not appear to be listening?
The ports don't show up when I execute `netstat -ab` in an admin command prompt. 
And if this is the cause - then of course, how do I fix this?

**Comment by ab-20-m** [comment]
You need to update the database.driver property in mirth.properties to use the class name of the SQL Server JDBC Driver. It should look something like:

database.driver = com.microsoft.sqlserver.jdbc.SQLServerDriver

**Reply by Coooz** [reply]
Hi ab-20-m,
No, that did not help. Thank you for suggesting, that was something I indeed had not tried yet.

**Comment by rjkroll** [comment]
Can you provide more details?   Are you using this connection string in the Source Transformer of a database reader?   Are you using it via javascript in a transformer step?    

Also, what copy of Microsoft Driver are you using?   I believe you would need at least v10.2 to work with Mirth 3.12

**Reply by Coooz** [reply]
Hi rjkroll,

"Are you using this connection string in the Source Transformer of a database reader? Are you using it via javascript in a transformer step?"
No, not at all, if only that were true I'd be much further by now... I need the correct url in the Database tab of the Server Manager. When you choose _sqlserver_ as Type, the default url is _jdbc:jtds:sqlserver://localhost:1433/mirthdb_. I was hoping this would suffice, but no.
Here I have to provide a username and password as well, and since I use Windows authentication, I filled in my Windows username and password. Should I leave those blank? I haven't tried that.
What I _have_ tried is specifying the domain with my username: I have to log in as XX\\yyyy. I have tried both including XX\\ and leaving this out.

I don't actually know what copy of Microsoft Driver I am using - where can I find this info?

As I am typing this, I see new logging that suggests that it may be a firewall issue. I've asked the customer whether this could be the case.
I keep my fingers crossed and will give an update asap. 

Thanks for contributing!

**Reply by rjkroll** [reply]
You can see what drivers you have here:  %ProgramFiles%\Mirth Connect\server-lib\database

[https://forums.mirthproject.io/forum/mirth-connect/support/180890-updating-microsoft-jdbc-driver-for-mirthdb](url)

**Comment by pacmano1** [comment]
See https://github.com/nextgenhealthcare/connect/discussions/5087, at the bottom. Also,  I think there are a handful of related threads I have commented on, most of which are already marked as answered.

@Coooz  post the full relevant portions of mirth.properties.

**Comment by Coooz** [comment]
Update: 

Our customer did not succeed in getting the connection working with Windows authentication either. 
He didn't delve any further in what might be the cause of that, but created a SQL account, which did the trick. It is working now.

Everyone who took the time and effort to contribute to this thread: thanks a lot! It is always good to know that I'm not on my own when problems emerge.
